### DISCUSSION TITLE
Considering add routers

### DISCUSSION THREAD

**Discussion by DennisZhangOiler** [slint-ui/slint#6783] [Ideas] [answered]
I found out that slint does not have router currently, which will be very convenient when developing multiple scene apps.

**Comment by ogoffart** [comment, marked as answer]
Could you elaborate on what exactly you mean by "router"?

In my mind, this means something that associate some particular URL to some particular ui component to render. 
But Slint is not a web framework and doesn't work with URL, so this is a bit abstract.

What is done if you have multiple scene is to have:
```slint
export enum Page { foo, bar }

export component MyWindow inherits Window {
   in-out property <Page> page;
   if page == Page.foo : FooPage { ... }
   if page == Page.bar : BarPage { ... }
}
```

**Comment by rizary** [comment]
I just got here, but I think if someone wants to build like slint library who handled routing like https://atomic-router.github.io/ is really good. I wonder if we can make it ourselves and treat it as some plugin into our app.

**Reply by javiercont97** [reply]
Hey, I've been working on a template for Slint on Android devices and implemented a stack-based navigation system. If you want to take a look go ahead https://github.com/javiercont97/android_template_slint_rust/tree/stack_based_navigation. It may help you and perhaps serve as inspiration. It uses the the same conditional rendering logic shown above but it keeps track of the pages and lets you navigate back.
